main/lsp/ExtractVariable.cc

```diff
@@ -45,6 +45,8 @@ core::LocOffsets findWhereToInsert(const ast::ExpressionPtr &scope, const core::
             whereToInsert = if_->thenp.loc();
         } else if (if_->elsep.loc().exists() && if_->elsep.loc().contains(target)) {
             whereToInsert = if_->elsep.loc();
+        } else if (if_->cond.loc().exists() && if_->cond.loc().contains(target)) {
+            ENFORCE(false, "shouldn't be inserting into the cond of an if");
```

Comment:
Minor, but you might want to move this into a `DEBUG_ONLY` or `if constexpr debugmode` condition in the `else` to be sure that we don't pay the cost of the `if` condition in release builds, because the `else` branch is also an `ENFORCE(false)`.

Reply:
I'll just do what you suggested for the `while` case (move the condition to the `ENFORCE`), there's no other case to handle now that we have an case for all three parts of the `if`.